### DISCUSSION TITLE
Is WindowManager a singleton?  If not, how to explicitly create it?

### DISCUSSION THREAD

**Discussion by mwdpb** [canton7/Stylet#315] [Q&A] [answered]
I understand that you can have access to WindowManager by making it an argument in the viewmodel's constructor, see below:

![image](https://user-images.githubusercontent.com/55845355/144101025-ce4b53da-3a57-43b3-844e-94886c81064e.png)

But other than this place, how do I get access to it?  Is it a singleton?  If not, how can I create it?

Thanks in advance!

**Comment by canton7** [comment, marked as answer]
It's got a [few dependencies of its own](https://github.com/canton7/Stylet/blob/5e97f9e1ea8c45e9d7033e6e7c7ea8ce41c3420a/Stylet/WindowManager.cs#L94), so its not a singleton, other than the fact it's registered as a singleton with the ioc container.

You could always grab the instance in your bootstrapper and store it somewhere:

```cs
protected override void Configure()
{
    SomeStaticClass.WindowManager = Container.Get<IWindowManager>();
}
```

**Reply by mwdpb** [reply]
That worked perfectly, thank you!
